libs/sdk/src/react/stream.tsx

```diff
@@ -314,14 +326,23 @@ function getBranchView<StateType extends Record<string, unknown>>(
 
 function fetchHistory<StateType extends Record<string, unknown>>(
   client: Client,
-  threadId: string
+  threadId: string,
+  options?: { limit?: boolean | number }
 ) {
-  return client.threads.getHistory<StateType>(threadId, { limit: 1000 });
+  if (options?.limit === false) {
+    return client.threads
+      .getState<StateType>(threadId)
+      .then((state) => [state]);
+  }
+
+  const limit = typeof options?.limit === "number" ? options.limit : 1000;
```

Comment:
should we also decrease the default? 1k is probably more than what people would want (?)

Reply:
Would like to leave it as-is for now, since this may break users who rely on large limit (ie. a graph with a lot of nodes executed in between forks)